lib/puppet/type/mount.rb

```diff
@@ -293,5 +297,16 @@ def value(name)
       dependencies[0..-2]
     end
 
+    # Autorequire the mount point's file resource
+    autorequire(:file) { Pathname.new(@parameters[:name].value) }
+
+    # Autobefore the mount point's child file paths
+    autobefore(:file) do
+      dependencies = []
+      Pathname.new(@parameters[:path].value).descend do |child|
```

Comment:
The spec tests are pointing out that `path` is nil (not set). Sounds like this should be using `name` instead.